File: scripts/create_test_users.py
#!/usr/bin/env python3
"""
Script para criar usuários de teste no Firebase Authentication e Firestore.
Uso: python scripts/create_test_users.py service-account.json
"""

import firebase_admin
from firebase_admin import credentials, auth, firestore
import sys
import json

def create_test_user(db, email, password, name, phone, role, gender=None):
    """Cria um usuário no Firebase Authentication e documento no Firestore."""
    try:
        # Criar usuário no Authentication
        user = auth.create_user(
            email=email,
            password=password,
            display_name=name
        )
        
        # Criar documento no Firestore
        user_data = {
            'name': name,
            'email': email,
            'phone': phone,
            'role': role,
            'photoUrl': None,
            'isActive': True,
            'createdAt': firestore.SERVER_TIMESTAMP
        }
        
        if gender:
            user_data['gender'] = gender
        
        db.collection('users').document(user.uid).set(user_data)
        
        print(f"✅ Usuário criado: {email}")
        print(f"   Nome: {name}")
        print(f"   Papel: {role}")
        print(f"   UID: {user.uid}")
        print()
        return user.uid
        
    except Exception as e:
        print(f"❌ Erro ao criar usuário {email}: {e}")
        return None

def main():
    if len(sys.argv) < 2:
        print("Uso: python create_test_users.py <service-account.json>")
        print("Obtenha a service account em: Firebase Console → Project Settings → Service Accounts")
        sys.exit(1)
    
    service_account_path = sys.argv[1]
    
    try:
        # Inicializar Firebase Admin SDK
        cred = credentials.Certificate(service_account_path)
        firebase_admin.initialize_app(cred)
        
        db = firestore.client()
        
        print("🔥 Criando usuários de teste no Firebase...")
        print("=" * 50)
        print()
        
        # Criar usuários de teste
        users = [
            {
                'email': 'cliente@teste.com',
                'password': 'cliente123',
                'name': 'Maria Silva',
                'phone': '(11) 99999-1001',
                'role': 'client',
                'gender': None
            },
            {
                'email': 'prestadora@teste.com',
                'password': 'prestadora123',
                'name': 'Ana Costa',
                'phone': '(11) 99999-1002',
                'role': 'provider',
                'gender': 'female'
            },
            {
                'email': 'prestadora2@teste.com',
                'password': 'prestadora123',
                'name': 'Juliana Santos',
                'phone': '(11) 99999-1004',
                'role': 'provider',
                'gender': 'female'
            },
            {
                'email': 'admin@teste.com',
                'password': 'admin123',
                'name': 'Administrador',
                'phone': '(11) 99999-1003',
                'role': 'admin',
                'gender': None
            }
        ]
        
        created_count = 0
        for user_data in users:
            uid = create_test_user(
                db=db,
                email=user_data['email'],
                password=user_data['password'],
                name=user_data['name'],
                phone=user_data['phone'],
                role=user_data['role'],
                gender=user_data['gender']
            )
            if uid:
                created_count += 1
        
        print("=" * 50)
        print(f"✅ {created_count} usuários de teste criados com sucesso!")
        print()
        print("📱 Credenciais de Login:")
        print("👤 Cliente: cliente@teste.com / cliente123")
        print("👩 Prestadora: prestadora@teste.com / prestadora123")
        print("👩 Prestadora 2: prestadora2@teste.com / prestadora123")
        print("👨‍💼 Admin: admin@teste.com / admin123")
        print("🔑 Código Admin: MARIAVAI-ADMIN-2026")
        
    except Exception as e:
        print(f"❌ Erro: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: scripts/create_test_users.dart
import 'package:firebase_admin/firebase_admin.dart';
import 'package:firebase_admin/src/auth.dart' as auth;
import 'package:firebase_admin/src/firestore.dart' as firestore;

/// Script para criar usuários de teste no Firebase
/// Uso: dart scripts/create_test_users.dart <service-account.json>
void main(List<String> args) async {
  if (args.isEmpty) {
    print('Uso: dart create_test_users.dart <service-account.json>');
    print('Obtenha a service account em: Firebase Console → Project Settings → Service Accounts');
    return;
  }

  final serviceAccountPath = args[0];

  try {
    // Inicializar Firebase Admin SDK
    final credentials = 
        ServiceAccountCredentials.fromServiceAccountFile(serviceAccountPath);
    await FirebaseAdmin.initializeApp(credentials);

    final db = firestore.Firestore.instance;

    print('🔥 Criando usuários de teste no Firebase...');
    print('=' * 50);
    print('');

    // Dados dos usuários de teste
    final testUsers = [
      {
        'email': 'cliente@teste.com',
        'password': 'cliente123',
        'name': 'Maria Silva',
        'phone': '(11) 99999-1001',
        'role': 'client',
        'gender': null,
      },
      {
        'email': 'prestadora@teste.com',
        'password': 'prestadora123',
        'name': 'Ana Costa',
        'phone': '(11) 99999-1002',
        'role': 'provider',
        'gender': 'female',
      },
      {
        'email': 'prestadora2@teste.com',
        'password': 'prestadora123',
        'name': 'Juliana Santos',
        'phone': '(11) 99999-1004',
        'role': 'provider',
        'gender': 'female',
      },
      {
        'email': 'admin@teste.com',
        'password': 'admin123',
        'name': 'Administrador',
        'phone': '(11) 99999-1003',
        'role': 'admin',
        'gender': null,
      },
    ];

    int createdCount = 0;

    for (final userData in testUsers) {
      try {
        // Criar usuário no Authentication
        final userRecord = await auth.Auth.instance.createUser(
          email: userData['email'] as String,
          password: userData['password'] as String,
          displayName: userData['name'] as String,
        );

        // Criar documento no Firestore
        final userDoc = db.collection('users').doc(userRecord.uid);
        final userDataMap = {
          'name': userData['name'],
          'email': userData['email'],
          'phone': userData['phone'],
          'role': userData['role'],
          'photoUrl': null,
          'isActive': true,
          'createdAt': firestore.FieldValue.serverTimestamp(),
        };

        if (userData['gender'] != null) {
          userDataMap['gender'] = userData['gender'];
        }

        await userDoc.set(userDataMap);

        print('✅ Usuário criado: ${userData['email']}');
        print('   Nome: ${userData['name']}');
        print('   Papel: ${userData['role']}');
        print('   UID: ${userRecord.uid}');
        print('');
        createdCount++;
      } catch (e) {
        print('❌ Erro ao criar usuário ${userData['email']}: $e');
        print('');
      }
    }

    print('=' * 50);
    print('✅ $createdCount usuários de teste criados com sucesso!');
    print('');
    print('📱 Credenciais de Login:');
    print('👤 Cliente: cliente@teste.com / cliente123');
    print('👩 Prestadora: prestadora@teste.com / prestadora123');
    print('👩 Prestadora 2: prestadora2@teste.com / prestadora123');
    print('👨‍💼 Admin: admin@teste.com / admin123');
    print('🔑 Código Admin: MARIAVAI-ADMIN-2026');

    await FirebaseAdmin.deleteApp();
  } catch (e) {
    print('❌ Erro: $e');
  }
}